### PR TITLE
Merge for device_exposure

### PR DESCRIPTION
**Pull request by yzLim12**

### CODE REVIEW

**Comment by yzLim12** [comment]
ETL development based on [https://github.com/sidataplus/PASAR_ETL_Spec/blob/v0.2.2-alpha/device_exposure.md](url)

Produced stg__device_exposure and int__device_exposure which before populating in device_exposure table. Please kindly review and do let me know if it's good to populate.

**Comment by hamcheezee** [comment]
LGTM

**Review by hamcheezee — APPROVED** [review]

File: etl/pypasar/omop/device_exposure.py [hunk 1]
```diff
@@ -22,14 +22,212 @@ def execute(self):
...
+                connection.execute(text("DROP VIEW IF EXISTS stg__device_exposure"))
+
+                # Delete device_exposure table
+                connection.execute(text("TRUNCATE TABLE device_exposure"))
 
     def process(self):
         # In batches
+        omop_schema = os.getenv("POSTGRES_OMOP_SCHEMA")
+        postop_schema = os.getenv("POSTOP_SOURCE_SCHEMA")
+        preop_schema = os.getenv("PREOP_SOURCE_SCHEMA")
```
Comment (satish-a0) [inline etl/pypasar/omop/device_exposure.py:41]: Please update to the existing env variables https://github.com/satish-a0/pasar/blob/main/etl/.env.example#L8-L10

File: etl/pypasar/omop/device_exposure.py [hunk 2]
```diff
@@ -22,14 +22,212 @@ def execute(self):
...
+                        SELECT
+                            -- Autogenerate unique device_exposure_id based on device_exposure_start_date and id
+                            ROW_NUMBER() OVER (ORDER BY device_exposure_start_date, id) AS device_exposure_id,
+                            person_id,
+                            device_concept_id,
+                            device_exposure_start_date,
+                            NULL AS device_exposure_start_datetime,
+                            device_exposure_end_date,
+                            NULL AS device_exposure_end_datetime,
+                            32879 AS device_type_concept_id, -- Registry concept id
```
Comment (satish-a0) [inline etl/pypasar/omop/device_exposure.py:217]: Added to clarifications

File: etl/pypasar/omop/device_exposure.py (hunk 1 above)
Comment (yzLim12) [reply etl/pypasar/omop/device_exposure.py]: Thanks for informing. Already updated.

**Review by satish-a0 — APPROVED** [review]

**Pull request merged**